pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -148,9 +180,54 @@ public void build()
       StarTreeIndexMapUtils
           .storeToFile(indexMaps, new File(_segmentDirectory, StarTreeV2Constants.INDEX_MAP_FILE_NAME));
       FileUtils.forceDelete(starTreeIndexDir);
+      _existingStarTrees.cleanOutputDirectory();
     }
 
-    LOGGER.info("Finished building {} star-trees in {}ms", numStarTrees, System.currentTimeMillis() - startTime);
+    LOGGER.info("Newly built start-trees {}.\n Reused star-trees {}.", numStarTrees - reusedStarTrees, reusedStarTrees);
+    LOGGER.info("Finished building star-trees in {}ms", System.currentTimeMillis() - startTime);
+  }
+
+  /**
+   * Extracts the individual star-trees from the combined index file and returns {@link SeparatedStarTreesMetadata}
+   * which contains the information about the output directory where the extracted star-trees are located and also
+   * has the builder configs of each of the star-tree.
+   */
+  private SeparatedStarTreesMetadata extractStarTrees()
+      throws Exception {
+    SeparatedStarTreesMetadata metadata = new SeparatedStarTreesMetadata(_segmentDirectory);
+    if (_metadataProperties.containsKey(MetadataKey.STAR_TREE_COUNT)) {
+      File indexFile = new File(_segmentDirectory, StarTreeV2Constants.INDEX_FILE_NAME);
+      File indexMapFile = new File(_segmentDirectory, StarTreeV2Constants.INDEX_MAP_FILE_NAME);
+      try (StarTreeIndexSeparator separator =
+          new StarTreeIndexSeparator(indexMapFile, indexFile,
+              _metadataProperties.getInt(MetadataKey.STAR_TREE_COUNT))) {
+        separator.separate(metadata.getOutputDirectory());
+        metadata.setBuilderConfigList(separator.extractBuilderConfigs(_metadataProperties));
```

Comment:
Consider making these fields read-only (set them in the constructor). `SeparatedStarTreesMetadata` should be a read-only helper class to extract fields from the metadata